helm/dagger/templates/engine-daemonset.yaml

```diff
@@ -70,7 +70,11 @@ spec:
           resources: {{- toYaml .Values.engine.resources | nindent 12 }}
           readinessProbe:
             exec:
-              command: ["buildctl", "debug", "workers"]
+              command:
+                - sh
+                - -exc
+                - |-
+                  dagger core container with-new-file --contents "We are ready" --path "/tmp/ready" file --path "/tmp/ready" contents
```

Comment:
This is a great, much needed change!

Instead of using `dagger core` which is unstable and likely to change - see https://github.com/dagger/dagger/pull/8135 - I propose that we use this command instead: `echo "{ directory { id } }" | dagger query`

FTR:
```console
❯ echo "{ directory { id } }" | time dagger query
Setup tracing at https://dagger.cloud/traces/setup. To hide: export SHUTUP=1

✔ connect 0.7s
✔ cache request: mkfile /schema.json 0.0s
✔ load cache: mkfile /schema.json 0.0s
✔ cache request: blob://sha256:b618d3d179857032b7a416ad037342be687116289c0fd1cc8b4a39a9f2d65d4e 0.0s
✔ load cache: blob://sha256:b618d3d179857032b7a416ad037342be687116289c0fd1cc8b4a39a9f2d65d4e 0.0s
✔ moduleSource(refString: "."): ModuleSource! 0.0s
✔ ModuleSource.kind: ModuleSourceKind! 0.0s
✔ directory: Directory! 0.0s

{
    "directory": {
        "id": "ChV4eGgzOmMwMmVlMmZiODlhYjRmMDQSSgoVeHhoMzpjMDJlZTJmYjg5YWI0ZjA0EjESDQoJRGlyZWN0b3J5GAEaCWRpcmVjdG9yeUoVeHhoMzpjMDJlZTJmYjg5YWI0ZjA0"
    }
}
dagger query  0.16s user 0.14s system 20% cpu 1.492 total
```

**After** `dagger core` stabilises, we can easily switch to it.

Reply:
I'd add to this - we don't even need to do a full operation at all, we could just do `{ version }` as a query. We're really just interested if the dagger server is running at all, I don't think we need to simulate an operation here?